test/listeners/upload.js

```diff
@@ -15,7 +15,8 @@ const sinon = require('sinon');
 const upload = require('../../src/listeners/upload');
```

Comment:
Tests to cover new creation assertions. Would have preferred if we could have figured out the onFirstCall and onSecondCall behaviors, but that's not well documented for sinon mocks. Thinking back, I've resorted to stubs in these sorts of cases and then worked out my own assertions for them, but this seems like it may be more straightfoward.